frontend/app/services/personal-information-service.server.ts

```diff
@@ -6,106 +6,113 @@ import { getLogger } from '~/utils/logging.server';
+        cityName: z.string(),
+        provinceName: z.string().optional(),
+        countryName: z.string(),
+        countryCode: z.string().optional(),
+        postalCode: z.string().optional(),
       })
-      .array()
       .optional(),
-    PersonContactInformation: z
+    mailingAddress: z
       .object({
-        EmailAddress: z.object({
-          EmailAddressID: z.string(),
-        }),
-        TelephoneNumber: z.object({
-          FullTelephoneNumber: z.string(),
-        }),
-        Address: z
-          .object({
-            AddressCategoryCode: z.object({
-              ReferenceDataName: z.string(),
-            }),
-            AddressStreet: z.object({
-              StreetName: z.string(),
-            }),
-            AddressSecondaryUnitText: z.string(),
-            AddressCityName: z.string(),
-            AddressProvince: z.object({
-              ProvinceName: z.string(),
-            }),
-            AddressCountry: z.object({
-              CountryName: z.string(),
-              CountryCode: z.string().optional(), //TODO ask if this field is a typo
-            }),
-            AddressPostalCode: z.string(),
-          })
-          .array(),
+        streetName: z.string(),
+        cityName: z.string(),
+        provinceName: z.string().optional(),
+        countryName: z.string(),
+        countryCode: z.string().optional(),
+        postalCode: z.string().optional(),
       })
       .optional(),
-    PersonLanguage: z
+    emailAddress: z.string().optional(),
+    phoneNumber: z.string().optional(),
+    language: z
       .object({
-        LanguageCode: z.object({
-          ReferenceDataName: z.string(),
-        }),
-        PreferredIndicator: z.boolean(),
+        languageCode: z.string().optional(),
+        preferredIndicator: z.boolean().optional(),
       })
       .optional(),
-    PersonSINIdentification: z.object({
-      IdentificationID: z.string(),
-    }),
-  }),
-});
-
-type PersonalInformationApiResponse = z.infer<typeof personalInformationApiResponseSchema>;
-
-const personalInfoDtoSchema = z.object({
-  applictantId: z.string().optional(),
-  clientId: z.string().optional(),
-  lastName: z.string().optional(),
-  firstName: z.string().optional(),
-  homeAddress: z
-    .object({
-      streetName: z.string(),
-      cityName: z.string(),
-      provinceName: z.string().optional(),
-      countryName: z.string(),
-      postalCode: z.string().optional(),
-    })
-    .optional(),
-  mailingAddress: z
-    .object({
-      streetName: z.string(),
-      cityName: z.string(),
-      provinceName: z.string().optional(),
-      countryName: z.string(),
-      postalCode: z.string().optional(),
-    })
-    .optional(),
-  emailAddress: z.string().optional(),
-  phoneNumber: z.string().optional(),
-  language: z
-    .object({
-      languageCode: z.string().optional(),
-      preferredIndicator: z.boolean().optional(),
-    })
-    .optional(),
-});
+  })
+  .optional();
```

Comment:
Similar comment as the one made on `personalInformationApiResponseSchema`.

Reply:
I understand for personalInformationApiResponseSchema but if we were to add the call on the frontend, wouldn't React complain it can be null? Olivier had mentioned running into it when adding it.
![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/156207329/fa2b76bc-a831-4a9a-8410-1694d7e44943)

Reply:
`getPersonalInformation(..)` currently returns `null` if the response is a 204 though. 

I can't tell from that screenshot where that code comes from, but could it be you're doing a not-null assertion on a promise? Maybe we're missing an `await` since `getPersonalInformation` is currently `async`?

```typescript
const personalInformation = await getPersonalInformation(sin)
if (!personalInformation) { throw redirect(..); }
```

Reply:
Removed: 102180defa666472647423336ecff5b8c2e4f0c2